apps/users/src/app/app.controller.ts

```diff
@@ -239,7 +241,17 @@ export class UserController {
   async getMyData(@Req() req: CredentialRequest) {
     //UserService
     const user = await this.userService.getUserFromCredential(req.$credential);
-    if (user) return await user.toUserResponse();
+    if (
+      user &&
+      req.params &&
+      req.params['userFields'] &&
+      req.params['userFields'] === 'wallet'
```

Comment:
```suggestion
      user && req.params?.['userFields'] === 'wallet'
```

Better to use the optional chaining